tool/tsh/db.go

```diff
@@ -658,11 +658,10 @@ type localProxyConfig struct {
 func prepareLocalProxyOptions(arg *localProxyConfig) (*localProxyOpts, error) {
 	certFile := arg.cliConf.LocalProxyCertFile
 	keyFile := arg.cliConf.LocalProxyKeyFile
-	if arg.routeToDatabase.Protocol == defaults.ProtocolSQLServer ||
-		arg.routeToDatabase.Protocol == defaults.ProtocolCassandra ||
-		(arg.localProxyTunnel && certFile == "") {
-		// For SQL Server and Cassandra connections, local proxy must be configured with the
-		// client certificate that will be used to route connections.
+	if arg.localProxyTunnel && (certFile == "" || keyFile == "") {
+		// --tunnel implies the local proxy is configured with db certs.
+		// If db certs are not specified but exist in the user profile, load
+		// them from the profile.
```

Comment:
I am wondering if we drop completely logic for `arg.cliConf.LocalProxyCertFile` and `arg.cliConf.LocalProxyKeyFile` here, will the middleware take care of the certs anyway?

I assume that the values users have to pass into `--key-file` and `--cert-file` should match what tunnel mode uses by default.

Reply:
`--tunnel` will take care of certs automatically anyway, but it doesn't check user profile for existing certs. We load them in here to avoid reissuing certs when we already have them in the profile

Reply:
yes you are right. i got confused too for a minute myself.